common/src/main/java/io/netty/util/internal/CleanerJava25Linker.java

```diff
@@ -0,0 +1,269 @@
+                    throw new UnsupportedOperationException(
+                            "Linking to malloc and free is only supported on 64-bit platforms.");
+                }
+
+                // Finally, we create three method handles, for malloc, free, and for wrapping an address in a
+                // ByteBuffer. Effectively, we need the equivalent of these three code snippets:
+                //
+                //        MemorySegment mallocPtr = Linker.nativeLinker().defaultLookup().find("malloc").get();
+                //        MethodHandle malloc = Linker.nativeLinker().downcallHandle(
+                //                mallocPtr, FunctionDescriptor.of(ValueLayout.JAVA_LONG, ValueLayout.JAVA_LONG));
+                //
+                //        MemorySegment freePtr = Linker.nativeLinker().defaultLookup().find("free").get();
+                //        MethodHandle free = Linker.nativeLinker().downcallHandle(
+                //                freePtr, FunctionDescriptor.ofVoid(ValueLayout.JAVA_LONG));
+                //
+                //        ByteBuffer byteBuffer = MemorySegment.ofAddress(addr).asByteBuffer();
+                Class<?> ofLongValueLayoutCls = Class.forName("java.lang.foreign.ValueLayout$OfLong");
+                Class<?> linkerCls = Class.forName("java.lang.foreign.Linker");
+                Class<?> linkerOptionCls = Class.forName("java.lang.foreign.Linker$Option");
+                Class<?> linkerOptionArrayCls = Class.forName("[Ljava.lang.foreign.Linker$Option;");
+                Class<?> symbolLookupCls = Class.forName("java.lang.foreign.SymbolLookup");
+                Class<?> optionalCls = Class.forName("java.util.Optional");
+                Class<?> memSegCls = Class.forName("java.lang.foreign.MemorySegment");
+                Class<?> funcDescCls = Class.forName("java.lang.foreign.FunctionDescriptor");
+
+                MethodHandle optGetMemSeg = lookup.findVirtual(optionalCls, "get", methodType(Object.class))
+                        .asType(methodType(memSegCls, optionalCls));
+                MethodHandle nativeLinker = lookup.findStatic(linkerCls, "nativeLinker", methodType(linkerCls));
+                MethodHandle defaultLookupStatic = MethodHandles.foldArguments(
+                        lookup.findVirtual(linkerCls, "defaultLookup", methodType(symbolLookupCls)),
+                        nativeLinker);
+                MethodHandle downcallHandleStatic = MethodHandles.foldArguments(
+                        lookup.findVirtual(linkerCls, "downcallHandle",
+                                methodType(MethodHandle.class, memSegCls, funcDescCls, linkerOptionArrayCls)),
+                        nativeLinker);
+                MethodHandle findSymbol = MethodHandles.filterReturnValue(
+                        MethodHandles.foldArguments(
+                                lookup.findVirtual(symbolLookupCls, "find", methodType(optionalCls, String.class)),
```

Comment:
As this is intended to run on java 25, maybe it's easier to use `findOrThrow` here instead, this would remove the need for the additional `Optional.get` call (`findOrThrow` is available since jdk 23 https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/lang/foreign/SymbolLookup.html#findOrThrow(java.lang.String))